services/app-web/src/components/FileUpload/FileUpload.tsx

```diff
@@ -270,13 +272,28 @@ export const FileUpload = ({
                         }
                     }
                 })
-                .catch((_e) => {
+                .catch(async (_e) => {
+                    const error = new Error(
+                        `UPLOAD_ERROR: ${JSON.stringify(
+                            _e
+                        )}. Files added: ${JSON.stringify(files)}`
+                    )
+                    recordJSException(error)
+
+                    // file upload failing could be due to session timeout that did not hit the modal
+                    // double check the user still has their session, if not, logout to update the React state with their login status
+                    try {
+                        await checkAuth()
```

Comment:
I'm still trying to test this out in the app,  but I was curious about why we check the session after a failed upload instead of checking before starting the upload.

Reply:
@JasonLin0991 To avoid delaying the start of file upload with another request is the reason I had.